pkg/kubernetes/nulecule.js

```diff
@@ -0,0 +1,257 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2014 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+define([
+    "jquery",
+    "base1/cockpit",
+    "kubernetes/config"
+], function($, cockpit, config) {
+    "use strict";
+
+    var nulecule = { };
+    var _ = cockpit.gettext;
+
+    function debug() {
+        if (window.debugging == "all" || window.debugging == "nulecule")
+            console.debug.apply(console, arguments);
+    }
+
+    function failure(ex) {
+        console.warn(ex);
+    }
+
+    /**
+     * NuleculeClient
+     *
+     */
+    function NuleculeClient() {
+        var self = this;
+        var http = cockpit.http(5000);
+        var version = '0.1.1';
+        var status = "";
+        var answers = {};
+        var install_dir = "";
+
+        self.create_tmp = function create_tmp() {
+            var process = cockpit.spawn(['/bin/sh', '-s']).input("mktemp -p /tmp -d APP_ENTITY.XXXX");
+            return process;
+        };
+
+        self.writeAnswerfile = function writeAnswerfile(install_dir, data) {
+            return cockpit.file(install_dir + "/answers.conf").replace(JSON.stringify(data));
```

Comment:
This leaks the cockpit.file() channel. The file needs to be close() once the replace is done.